RealTimeFaceMaskDetector/TCPServer/SocketServer.cpp

```diff
@@ -58,37 +65,90 @@ bool SocketServer::BindListeningSocket()
 
 bool SocketServer::AcceptConnection()
 {
-	m_client_socket = accept(m_listen_socket, NULL, NULL);
-	if (m_client_socket == INVALID_SOCKET)
+	std::cout << "Waiting connection" << std::endl;
+	//Here make async wait
+	bool ready = false, stop = false;
+	SOCKET mock_socket = INVALID_SOCKET;
+	std::future<SOCKET> cl_socket = std::async(std::launch::async, [this,&ready] 
+		{ 
+			SOCKET result;
+
+			result = accept(m_listen_socket, NULL, NULL);
+
+			ready = true;
+			return result;
+		});
+	
+	std::cout << "Waiting...     Press \'ESC\' to stop the Server\n" << std::flush;
```

Comment:
If the server will run as a service, this output will not be written. Try to use Loger)

Reply:
most likely this feature will be deprecated